controllers/servingruntime_controller.go

```diff
@@ -286,6 +301,50 @@ func (r *ServingRuntimeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{RequeueAfter: requeueDuration}, nil
 }
 
+func (r *ServingRuntimeReconciler) getPVCs(ctx context.Context, req ctrl.Request) ([]string, error) {
+	s := &corev1.Secret{}
+	if err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      modelmesh.StorageSecretName,
+		Namespace: req.Namespace,
+	}, s); err != nil {
+		return nil, fmt.Errorf("Could not get the storage secret: %w", err)
+	}
+
+	pvcsMap := make(map[string]struct{})
+	var storageConfig map[string]string
+	for _, storageData := range s.Data {
+		if err := json.Unmarshal(storageData, &storageConfig); err != nil {
+			return nil, fmt.Errorf("Could not parse storage configuration json: %w", err)
+		}
+		if storageConfig["type"] == StoragePVCType {
+			pvcsMap[storageConfig["name"]] = struct{}{}
+		}
+	}
+	// add pvcs for predictors when the global config is enabled
+	if r.DeployPVCForPredictor {
+		list := &v1beta1.InferenceServiceList{}
```

Comment:
I think we still need to change this per [prior comment](https://github.com/kserve/modelmesh-serving/pull/267#discussion_r1020613091), to support different predictor "sources" (could be predictor or inferencservice). I will aim to write an example of what I mean.

Reply:
@chinhuang007 here is what I was thinking of, sorry again for the delay:
```go
	if cfg.AllowAnyPVC {
		restProxyEnabled := cfg.RESTProxy.Enabled
		f := func(p *api.Predictor) bool {
			if runtimeSupportsPredictor(rt, p, restProxyEnabled, req.Name) &&
				p.Spec.Storage != nil && p.Spec.Storage.Parameters != nil {
				
				params := *p.Spec.Storage.Parameters
				if stype := params["type"]; stype == "pvc" {
					if name := params["name"]; name != "" {
						pvcsMap[name] = struct{}{}
					}
				}
			}
			return false
		}

		for _, pr := range r.RegistryMap {
			if found, err := pr.Find(ctx, req.Namespace, f); found || err != nil {
				return nil, err
			}
		}
	}
```